samples/android/UrbanAirshipListener.cs

```diff
@@ -68,6 +69,31 @@ public void OnNotificationDismissed(NotificationInfo notificationInfo)
 		{
 			Log.Info (TAG, "Notification dismissed. Alert: " + notificationInfo.Message.Alert + ". Notification ID: " + notificationInfo.NotificationId);
 		}
-	}
+
+        public bool OnShowMessageCenter(String messageId)
+        {
+            using (var h = new Handler(Looper.MainLooper))
```

Comment:
Pretty sure this already runs on the main thread but its not annotated to do that. We should fix that in the next SDK release